### PR TITLE
Feature/iterator

### PR DESCRIPTION
**Pull request by ktarbet**

### CODE REVIEW

File: src/main/java/hec/ensemble/EnsembleUtility.java [hunk 1]
```diff
@@ -94,17 +94,20 @@ static void readModifyWrite(String fileName) throws Exception {
         int count = 0;
```
Comment (HenryGeorgist) [inline src/main/java/hec/ensemble/EnsembleUtility.java]: Can you tell me the difference in the timing between the previous set of code before iterator, ensembletimeseriesreader implementation of iterator, and the ensembletimeseries iterator for this test?

File: src/test/java/hec/ensemble/ResSimTest.java [hunk 2]
```diff
@@ -24,7 +24,7 @@ public void simulateResSim() throws Exception {
 
             TimeSeriesIdentifier tsid = new TimeSeriesIdentifier("Coyote.fake_forecast", "flow");
 
-            EnsembleTimeSeriesReader reader = db.getEnsembleTimeSeriesReader(tsid);
+            EnsembleTimeSeries reader = db.getEnsembleTimeSeries(tsid);
```
Comment (HenryGeorgist) [inline src/test/java/hec/ensemble/ResSimTest.java]: the name "reader" is confusing given the change in the structure of the code, consider renaming that local variable.

File: src/main/java/hec/ensemble/EnsembleUtility.java (hunk 1 above)
Comment (ktarbet) [reply src/main/java/hec/ensemble/EnsembleUtility.java]: commit|read(s)|write(s)|
--|--|-|
before |1.5| 5.7 |
lazy reader| 6| 5.8 |  
now | 1.5| 5.8 |

Appveyor numbers can vary , perhaps based on server load.  The write time changed from 7.3 to 5.8 with just a rebuild. Lazy reader numbers are from my memory.

Comment (HenryGeorgist) [reply src/main/java/hec/ensemble/EnsembleUtility.java]: this is good enough for this situation. obviously the efficency of reading the entire dataset with one transaction is worth it.

**Pull request merged**